### PR TITLE
Mike/6208 add hiv aoe

### PR DESCRIPTION
**Pull request by mpbrown**
# FRONTEND PULL REQUEST

## Related Issue

- #6208

## Changes Proposed

- Adds support for HIV AOE questions on the frontend
- If a positive result is selected, the AOE questions become visible and required

## Additional Information

- Depends on #6896 being merged first

## Testing

- Deployed on [dev2](https://dev2.simplereport.gov/app/settings/organization?facility=92a06f64-05c9-42e1-b984-152930de5b66)
- Organization **Apples** and testing facility **Another Testing Facility** have been configured with an HIV device

### HIV device selected with blank results

Screenshot 2023-11-13 160406

### Selecting positive causes the AOE questions to appear and be required
Screenshot 2023-11-13 160538

### Attempting to submit without answering the AOE questions will show error state
Screenshot 2023-11-13 160525
Screenshot 2023-11-13 160559

### Prefer not to answer is a valid response

Screenshot 2023-11-13 160616

### CODE REVIEW

**Comment by fzhao99** [comment]
Functionality and code look good overall to me! The questionnaire doesn't seem to match Figma designs though, so double checking that we got a design signoff? 

Happy to approve once confirmed

**Comment by mpbrown** [comment]
> Functionality and code look good overall to me! The questionnaire doesn't seem to match Figma designs though, so double checking that we got a design signoff?

Specifically the missing notes field that's shown on the Figma? I checked with Jayna and she said we can leave that off for now because design was still testing it out with users and the ticket only includes the AOE questions for pregnancy and sexual partners

File: frontend/src/app/testQueue/TestCardForm/diseaseSpecificComponents/HIVAoEForm.tsx
Comment (emyl3) [inline frontend/src/app/testQueue/TestCardForm/diseaseSpecificComponents/HIVAoEForm.tsx:72]: this class is causing the multiselect to change with more/less selections

https://github.com/CDCgov/prime-simplereport/assets/20211771/db04bdbe-4160-41d2-8278-c13d1a6dfe21

**Review by emyl3 — COMMENTED** [review]
Thank you for your work on this, Mike! 😸 Left one styling issue I found.

@mehansen @jayna-SkylightDigital 
I noticed  while testing this that our Print Modal, Download test results, and View details does not support HIV results. Do we have tickets for this? 🤔 I wasn't able to find anything regarding those changes in our backlog/issues.

**Comment by fzhao99** [comment]
> > Functionality and code look good overall to me! The questionnaire doesn't seem to match Figma designs though, so double checking that we got a design signoff?
> 
> Specifically the missing notes field that's shown on the Figma? I checked with Jayna and she said we can leave that off for now because design was still testing it out with users and the ticket only includes the AOE questions for pregnancy and sexual partners

Thanks for that context on the notes field! That makes sense. 

The other thing (at least that I can see) is it seems that the designs use a combobox / dropdown here rather than a multiselect. Those two components are pretty different, so just wanted to make sure those are aligned. 
Screenshot 2023-11-15 at 12 23 25 PM

Might be helpful just to get a general review from the designers in case anything else looks off.

File: frontend/src/app/testQueue/TestCardForm/diseaseSpecificComponents/HIVAoEForm.tsx
Comment (mpbrown) [reply frontend/src/app/testQueue/TestCardForm/diseaseSpecificComponents/HIVAoEForm.tsx:72]: @emyl3 thanks for finding this! I'm just waiting to hear back from Ryan regarding the desired sizing behavior of multiselect here

Comment (mpbrown) [reply frontend/src/app/testQueue/TestCardForm/diseaseSpecificComponents/HIVAoEForm.tsx:72]: @mehansen @jayna-SkylightDigital similar to what @emyl3 mentioned above, I noticed that those other parts of the app like Print and View details don't seem to show RSV either so we may need to prioritize a ticket for that to release RSV asap

Screenshot 2023-11-16 095000

Screenshot 2023-11-16 095028

Screenshot 2023-11-16 095050

**Comment by mpbrown** [comment]
Ryan and Tomi both tested and approved from the design/UX side. We updated the multiselect width although this might change in the future once we refine #6891 since the future proposed design is bringing test date/time, device dropdown, and specimen dropdown all onto the same row

**Review by fzhao99 — DISMISSED** [review]
lgtm! thanks for herding all the approvals :)

**Review by emyl3 — APPROVED** [review]
LGTM! Thank you Mike!! 🎉

**Review by fzhao99 — APPROVED** [review]

**Pull request merged**
